enforcer/utils/packetgen/packet_gen.go

```diff
@@ -0,0 +1,370 @@
+//Package packetgen is a Packet Generator library
+//This is a beta version which returns only TCP 3-way handshake
+//Updates are coming soon
+package packetgen
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+//NewPacket returns a packet strut which implements PacketManipulator
+func NewPacket() PacketManipulator {
+
+	return &Packet{}
+
+}
+
+//AddIPLayer creates an IP packet
+func (p *Packet) AddIPLayer(srcIPstr string, dstIPstr string) error {
+
+	if p.IPLayer != nil {
+		return fmt.Errorf("IP Layer already exists")
+	}
+
+	var srcIP, dstIP net.IP
+
+	//IP address of the source
+	srcIP = net.ParseIP(srcIPstr)
+
+	if srcIP == nil {
+		return fmt.Errorf("No source IP given")
+	}
+
+	//IP address of the destination
+	dstIP = net.ParseIP(dstIPstr)
+
+	if dstIP == nil {
+		return fmt.Errorf("No destination IP given")
+	}
+
+	//IP packet header
+	p.IPLayer = &layers.IPv4{
+		SrcIP:    srcIP,
+		DstIP:    dstIP,
+		Version:  4,
+		TTL:      64,
+		Protocol: layers.IPProtocolTCP,
+	}
+
+	return nil
+
+}
+
+//GetIPChecksum returns IP cheksum
+func (p *Packet) GetIPChecksum() uint16 {
+
+	return p.IPLayer.Checksum
+
+}
+
+//AddTCPLayer creates a TCP packet
+func (p *Packet) AddTCPLayer(srcPort layers.TCPPort, dstPort layers.TCPPort) error {
+
+	if p.TCPLayer != nil {
+		return fmt.Errorf("TCP Layer already exists")
+	}
+
+	//TCP packet header
+	p.TCPLayer = &layers.TCP{
+		SrcPort: srcPort,
+		DstPort: dstPort,
+		Window:  1505,
+		Urgent:  0,
+		Seq:     11050,
+		Ack:     0,
+		ACK:     false,
+		SYN:     false,
+		FIN:     false,
+		RST:     false,
+		URG:     false,
+		ECE:     false,
+		CWR:     false,
+		NS:      false,
+		PSH:     false,
+	}
+
+	p.TCPLayer.SetNetworkLayerForChecksum(p.IPLayer)
+
+	return nil
+
+}
+
+//SetTCPSequenceNumber changes TCP sequence number
+func (p *Packet) SetTCPSequenceNumber(seqNum uint32) error {
+
+	if p.TCPLayer.Seq != 0 {
+		return fmt.Errorf("Sequence number already exists")
+	}
+
+	p.TCPLayer.Seq = seqNum
+
```

Comment:
Thats good. So make sure that when someone asks for GetTCPPacket or GetTCPChecksum or ToBytes gets called the checksum is updated.